lightning/src/chain/keysinterface.rs

```diff
@@ -541,6 +541,8 @@ impl InMemorySigner {
 		let sighash = hash_to_message!(&bip143::SigHashCache::new(spend_tx).signature_hash(input_idx, &witness_script, descriptor.output.value, SigHashType::All)[..]);
 		let remotesig = secp_ctx.sign(&sighash, &self.payment_key);
 
+		if descriptor.output.script_pubkey != witness_script { return Err(()); }
```

Comment:
The witness script here is different from the output. You need to create a second address with `Address::p2wpkh` to compare against instead.